example/storage-provider-secrets/00-azure-blob-storage-secret.yaml

```diff
@@ -7,6 +7,7 @@ type: Opaque
 data: 
   storageAccount: YWRtaW4= # admin
   storageKey: YWRtaW4= # admin
+  # uri: blob.core.windows.net # override the ABS URI
```

Comment:
I will make it consistent and call it `storageAPIEndpoint`. Thanks for the feedback.

Reply:
I would differ here. I thought about this point, and realised that there's no real benefit in harmonizing the field name across providers, since we anyway already have different fields in the provider secret for the credentials. I would recommend using `uri` for Azure, since Azure uses this terminology, and leave it `storageAPIEndpoint` for Google. Any user who's trying to set this field will already have an idea about the provider and what the field itself means.

Reply:
Ok, thinking on this again, I realise that Google calls these endpoints `regional endpoints` ([ref](https://cloud.google.com/storage/docs/regional-endpoints)). And `storageAPIEndpoint` was a term we came up with in etcdbr. So maybe we need to re-look at this field name as well, or else harmonize the existing name across providers.

Reply:
Azure uses the terminology URI for `*.blob.core.windows.net` and `*.blob.core.chinacloudapi.cn` (the `*`, being key here). As @unmarshall mentions in this https://github.com/gardener/etcd-backup-restore/pull/756#discussion_r1696619193, this would be semantically correct since these are generic descriptors to URIs that can be formed for Azure resources.

However, using a field called `uri` for only the `blob.core.windows.net` part of the domain would be confusing since it is not a URI to the blob Azure resource/storage account.

Reply:
Yes calling only `blob.core.windows.net`  as URI is semantically and completely incorrect as per the definition of URI  as per [rfc3986](https://datatracker.ietf.org/doc/html/rfc3986#section-1.1.3).  This is a part of the subdomain. Therefore asking the user for a complete URI and then calling that is fine but asking the user to only specify a part of a subdomain and calling it a URI is not correct.

Reply:
Ok, makes sense. But would calling it `storageAPIEndpoint` be semantically correct, because that still refers to the entire endpoint, and not just part of it?

Reply:
The code has been changed to expect the full endpoint for the blob storage service, `http[s]://<storage-account-name>.<domain>` which is also the case in #691.

Is that what you mean?

Reply:
Ah ok, then perfect. We can go ahead and use any name that mentions it's an `endpoint` or `URL`.

Reply:
now its been changed to `domain`.